labs/lab-2-javascript-game/script-pt-3.js

```diff
@@ -0,0 +1,99 @@
+// debugger;
+
+var enemy;
+var user;
+var userHealthPercent;
+var enemyHealthPercent;
+
+window.addEventListener("load", () => {
+  document.getElementById("game-console").style.display = 'none';
+  document.getElementById("start-button").addEventListener("click", startGame);
+});
+
+class Character {
+  constructor(name, health, healsRemaining, winCount) {
+    this.name = name;
+    this.health = health;
+    this.healsRemaining = healsRemaining;
+    this.winCount = winCount;
+  }
+
+  generateAttackDamage() { // do min and max parameters / arguments
+    const min = 1;
+    const max = 3;
+    return (Math.floor(Math.random() * (max - min + 1) + min));
+  }
+
+  heal() {
+    const min = 1;
+    const max = 10;
+    return (Math.floor(Math.random() * (max - min + 1) + min));
+  }
+}
+
+function startGame() {
+  document.getElementById("game-console").style.display = 'block';
+  document.getElementById("start-container").style.display = 'none';
+  user = new Character("Bree", 40, 2, 0); // Have user be able to enter this
+  enemy = new Character("The Almighty Grant", 10);
+  document.getElementById("attack-button").addEventListener("click", startCombat);
+  document.getElementById("heal-button").addEventListener("click", userHeal);
+  document.getElementById("quit-button").addEventListener("click", quitGame);
+}
+
+function startCombat() {
+  user.health = user.health - enemy.generateAttackDamage();
+  enemy.health = enemy.health - user.generateAttackDamage();
+  percentHealthRemaining();
+  document.getElementById("user-health-bar").style.width = userHealthPercent;
+  document.getElementById("enemy-health-bar").style.width = enemyHealthPercent;
+  document.getElementById("user-health-number").innerText = user.health + "/40";
+  document.getElementById("enemy-health-number").innerText = enemy.health + "/10";
+  checkWins();
+}
+
+function percentHealthRemaining() {
+  userHealthPercent = ((user.health / 40) * 100) + "%";
+  enemyHealthPercent = ((enemy.health / 10) * 100) + "%";
+  return userHealthPercent;
+  return enemyHealthPercent;
```

Comment:
You can't return two things, unfortunately. 

I'd much rather see this be in the class. Then you could ask for `user.healthRemaining()` and `enemy.healthRemaining()`, and not do wild global variable passabouts.